bang/magicollections.py

```diff
@@ -2524,38 +2478,64 @@ def to_fields(self, view, item, extra_fields=None, exclude_fields=None, order=No
+                    'image': staticImageURL('language/{}.png'.format(_info['image'])),
+                    'link': getattr(item, '{}image_url'.format(_info['prefix'])),
+                    'link_text': string_concat(_('Image'), ' (', _info['translation'], ')'),
+                    'verbose_name': _('Image'),
+                    'verbose_name_subtitle': _info['translation'],
+                    'value': getattr(item, '{}image_thumbnail_url'.format(_info['prefix'])),
+                    'type': 'image_link',
+                }))
+
+        # Song + Event
+        for _field, _tl in [('song', _('Song')), ('event', _('Event'))]:
+            if getattr(item, _field):
+                 extra_fields.append((_field, subtitledImageLink(getattr(item, _field), _tl, _field)))
+            
         fields = super(AssetCollection, self).to_fields(view, item, *args, icons=icons, images={
             'image': staticImageURL('language/ja.png'),
             'english_image': staticImageURL('language/world.png'),
             'taiwanese_image': staticImageURL('language/zh-hant.png'),
             'korean_image': staticImageURL('language/kr.png'),
         }, extra_fields=extra_fields, exclude_fields=exclude_fields, order=order, **kwargs)
-        setSubField(fields, 'band', key='type', value=lambda f: 'image_link')
-        setSubField(fields, 'band', key='link', value=lambda f: u'/members/?i_band={}'.format(item.i_band))
-        setSubField(fields, 'band', key='ajax_link', value=lambda f: u'/ajax/members/?i_band={}&ajax_modal_only'.format(item.i_band))
-        setSubField(fields, 'band', key='link_text', value=lambda f: item.band)
-        setSubField(fields, 'band', key='value', value=lambda f: '{}img/band/{}.png'.format(RAW_CONTEXT['static_url'], item.band))
+
+        if 'band' in fields:
+            fields['band'] = bandField(item.band, item.i_band)
+
+        for _field, _tl in [('song', _('Song')), ('event', _('Event'))]:
+            if _field in fields:
+                 fields[_field] = subtitledImageLink(getattr(item, _field), _tl, _field)
+
+        if item.type is 'title' and item.value:
+            setSubField(fields, 'name', key='value', value=string_concat(item.t_name, ' ', item.value))
+        
         if item.source and item.source_link:
             setSubField(fields, 'source', key='type', value='link')
             setSubField(fields, 'source', key='value', value=item.source_link)
             setSubField(fields, 'source', key='icon', value='link')
             setSubField(fields, 'source', key='link_text', value=item.source)
+
+        if item.c_tags is not None:
+            asset_tags = '<div class="text-muted">'
+            for tag, _tl in models.Asset.TAGS_CHOICES:
+                if tag in item.c_tags and tag not in asset_tags:
+                    asset_tags += u'<a class="a-nodifference" href="/assets/?c_tags={}" data-ajax-url="/ajax/assets/?c_tags={}">#{}</a> '.format(tag, tag, unicode(_tl))
```

Comment:
This allows arbitrary HTML injection from translations, from [django's docs you should use format_html](https://docs.djangoproject.com/en/2.1/ref/utils/#django.utils.html.format_html) to safely format this.

(To test, translate one of the tag names to `"<script>alert('Hello')</script>"` and see if anything happens)